modules/codegen2/src/main/java/org/apache/ignite/internal/MessageSerializerGenerator.java

```diff
@@ -388,7 +388,11 @@ else if (assignableFrom(erasedType(type), type(Collection.class.getName()))) {
 
                 imports.add("org.apache.ignite.plugin.extensions.communication.MessageCollectionItemType");
 
-                returnFalseIfWriteFailed(write, "writer.writeCollection", getExpr,
+                String collectionWriter = assignableFrom(erasedType(type), type(Set.class.getName()))
```

Comment:
Let's keep simple `#writeCollection`. Currently, there are no difference, which collection is written.